integration/spark/src/main/common/java/io/openlineage/spark/agent/facets/builder/EnvironmentFacetBuilder.java

```diff
@@ -0,0 +1,108 @@
+package io.openlineage.spark.agent.facets.builder;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.type.CollectionType;
+import io.openlineage.spark.agent.facets.EnvironmentFacet;
+import io.openlineage.spark.agent.models.DatabricksMountpoint;
+import io.openlineage.spark.api.CustomFacetBuilder;
+import io.openlineage.spark.api.OpenLineageContext;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.function.BiConsumer;
+import org.apache.hc.client5.http.classic.methods.HttpPost;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpClient;
+import org.apache.hc.client5.http.impl.classic.CloseableHttpResponse;
+import org.apache.hc.client5.http.impl.classic.HttpClients;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+import org.apache.spark.scheduler.SparkListenerEvent;
+
+public class EnvironmentFacetBuilder
```

Comment:
BTW, since this implementation is very specific to the Databricks runtime, I'd rename this to `DatabricksEnvironmentFacetBuilder` - the facet itself is pretty generic and can be reused in other environments, but the builder class name should reflect the environment it's dependent on.